src/main/java/edu/hm/hafner/grading/MetricStatistics.java

```diff
@@ -62,19 +100,40 @@ public MetricStatistics add(final Value value, final String id) {
      *         if the metric is not available
      */
     public double asDouble(final String id) {
-        if (!projectValues.containsKey(id)) {
+        return this.asDouble(id, Scope.PROJECT);
+    }
+
+    /**
+     * Returns the metric value as double value.
+     *
+     * @param id
+     *         the metric id
+     * @param scope
+     *         the scope of the metric
+     *
+     * @return the metric value
+     * @throws IllegalArgumentException
+     *         if the metric is not available
+     */
+    public double asDouble(final String id, final Scope scope) {
+        var values = projectValues.computeIfAbsent(scope, b -> new HashMap<>());
```

Comment:
The code should be moved to a method (e.g., `getValues`) and reused in all duplications.